sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -251,27 +253,31 @@ export class EducationProgramOfferingController extends BaseController {
    * access to their specific offerings only).
    * @param locationId location id.
    * @param programId program id.
+   * @param programYearId program year id.
+   * @param selectedOfferingIntensity offering intensity selected by student.
    * @query includeInActivePY, if includeInActivePY is true,
    * then consider both active and inactive program year.
    * @returns key/value pair list of programs for students.
    */
   @AllowAuthorizedParty(AuthorizedParties.institution)
   @HasLocationAccess("locationId")
   @Get(
-    "location/:locationId/education-program/:programId/program-year/:programYearId/offerings-list",
+    "location/:locationId/education-program/:programId/program-year/:programYearId/selectedOfferingIntensity/:selectedOfferingIntensity/offerings-list",
   )
   async getProgramOfferingsForLocationForInstitution(
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
     @Param("programYearId") programYearId: number,
+    @Param("selectedOfferingIntensity")
+    selectedOfferingIntensity: OfferingIntensity,
     @Query("includeInActivePY") includeInActivePY = false,
   ): Promise<OptionItem[]> {
     const offerings =
       await this.programOfferingService.getProgramOfferingsForLocation(
         locationId,
         programId,
         programYearId,
-        undefined,
+        selectedOfferingIntensity,
```

Comment:
The parameter must be validated to ensure that is one of the valid enum options, as in the below example:
```typescript
if (
      !Object.values(DesignationAgreementStatus).includes(designationStatus)
    ) {
      throw new NotFoundException("Invalid designation agreement status.");
    }
```